src/box/memtx_engine.cc

```diff
@@ -191,6 +197,10 @@ memtx_build_secondary_keys(struct space *space, void *param)
 		}
 
 		for (uint32_t j = 1; j < space->index_count; j++) {
+			struct memtx_index *index =
+				(struct memtx_index *)space->index[j];
+			if (index->built_presorted)
```

Comment:
This condition would be required if we could specify whether to write sort data of specific spaces, but the functionality is absent, so, maybe, it's better to get rid of the extra variable and just check if the sort data is enabled and the index supports it (it must be written to the snapshot and thus, built using the O(n) algorithm at this point if the two conditions are true).

Reply:
The condition can also be useful if the space has a `before_replace` trigger (so we didn't use sort data for it).